From bd4daef404a21103848d7a28c15fc9895f96dfd5 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Fri, 7 Aug 2015 10:37:50 -0500
Subject: [PATCH 0232/5242] MLK-11336: mtd: nand: enable LPSR for GPMI NAND

commit  fcbfe6b1b171dd79c76654c4860fac021611c7f7 from
https://source.codeaurora.org/external/imx/linux-imx.git

The LPSR turns off the power for IOMUX when suspending so restore the
IOMUX when resuming in GPMI NAND driver.

The function was not tested yet since NAND only supported on 19x19
LPDDR board.

Signed-off-by: Han Xu <b45815@freescale.com>
Signed-off-by: Fugang Duan <B38611@freescale.com>
(cherry picked from commit: b0375f42a27044667082e53449e534b265d7a029)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
index 5f40e49..e99895e 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
@@ -2082,6 +2082,7 @@ static int gpmi_pm_suspend(struct device *dev)
 	struct gpmi_nand_data *this = dev_get_drvdata(dev);
 
 	release_dma_channels(this);
+	pinctrl_pm_select_sleep_state(dev);
 	return 0;
 }
 
@@ -2090,6 +2091,8 @@ static int gpmi_pm_resume(struct device *dev)
 	struct gpmi_nand_data *this = dev_get_drvdata(dev);
 	int ret;
 
+	pinctrl_pm_select_default_state(dev);
+
 	ret = acquire_dma_channels(this);
 	if (ret < 0)
 		return ret;
-- 
1.7.9.5

